python/lsst/afw/math/warpExposure.cc

```diff
@@ -20,109 +20,199 @@
+                "destImage"_a, "destWcs"_a, "srcImage"_a, "srcWcs"_a, "control"_a, "padValue"_a=EdgePixel);
+
+        mod.def("warpImage",
+                (int (*)(DestImageT &,
+                      SrcImageT const &,
+                      geom::XYTransform const &,
+                      WarpingControl const &,
+                      typename DestImageT::SinglePixel)) &warpImage<DestImageT, SrcImageT>,
+                "destImage"_a, "srcImage"_a, "xyTransform"_a, "control"_a, "padValue"_a=EdgePixel);
+
+        mod.def("warpCenteredImage",
+                &warpCenteredImage<DestImageT, SrcImageT>,
+                "destImage"_a, "srcImage"_a, "linearTransform"_a, "centerPoint"_a, "control"_a,
+                "padValue"_a=EdgePixel);
+    }
+
+    /**
+    Declare wrappers for warpExposure, warpImage and warpCenteredImage
+    for a particular pair of source and destination pixel types.
+
+    Declares both image and masked image variants of warpImage and warpCenteredImage.
+
+    @tparam DestPixelT  Desination pixel type, e.g. `int` or `float`
+    @tparam SrcPixelT  Source pixel type, e.g. `int` or `float`
+    @param[in,out] mod  pybind11 module for which to declare the function wrappers
+    */
+    template<typename DestPixelT, typename SrcPixelT> void declareWarpingFunctions(py::module & mod) {
+        using DestExposureT = image::Exposure<DestPixelT, image::MaskPixel, image::VariancePixel>;
+        using SrcExposureT = image::Exposure<SrcPixelT, image::MaskPixel, image::VariancePixel>;
+        using DestImageT = image::Image<DestPixelT>;
+        using SrcImageT = image::Image<SrcPixelT>;
+        using DestMaskedImageT = image::MaskedImage<DestPixelT, image::MaskPixel, image::VariancePixel>;
+        using SrcMaskedImageT = image::MaskedImage<SrcPixelT, image::MaskPixel, image::VariancePixel>;
+
+        mod.def("warpExposure",
+                &warpExposure<DestExposureT, SrcExposureT>,
+                "destExposure"_a,
+                "srcExposure"_a,
+                "control"_a,
+                "padValue"_a=edgePixel<DestMaskedImageT>(
+                    typename image::detail::image_traits<DestMaskedImageT>::image_category()));
+
+        declareImageWarpingFunctions<DestImageT, SrcImageT>(mod);
+        declareImageWarpingFunctions<DestMaskedImageT, SrcMaskedImageT>(mod);
+    }
 }
 
 PYBIND11_PLUGIN(_warpExposure) {
     py::module mod("_warpExposure", "Python wrapper for afw _warpExposure library");
 
-    py::class_<WarpingControl, std::shared_ptr<WarpingControl>> clsWarpingControl(mod, "WarpingControl");
+    /* Module level */
+    auto clsLanczosWarpingKernel = declareWarpingKernel<LanczosWarpingKernel>(mod, "LanczosWarpingKernel");
+    declareSimpleWarpingKernel<BilinearWarpingKernel>(mod, "BilinearWarpingKernel");
+    declareSimpleWarpingKernel<NearestWarpingKernel>(mod, "NearestWarpingKernel");
+
+    py::class_<WarpingControl> clsWarpingControl(mod, "WarpingControl");
```

Comment:
From the looks of the deleted line 
``py::class_<WarpingControl, std::shared_ptr<WarpingControl>> clsWarpingControl(mod, "WarpingControl");``,
 it looks like this needs at std::shared_ptr